### DISCUSSION TITLE
Extend Source Options for Loop Grid and Loop Carousel Widgets

### DISCUSSION THREAD

**Discussion by ayoubkhan558** [elementor/elementor#22170] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Hello Elementor team,  

Currently, the Loop Grid and Loop Carousel widgets allow us to loop over posts and post taxonomies. However, creating custom post types and fields for additional data sources can be cumbersome and time-consuming, especially for simple use cases like galleries, menus, or external data.  

To address these limitations, I propose extending the supported data sources to include:  
- **Posts**  
- **Post Taxonomies**  
- **Galleries**  
- **Items/Repeater Fields**  
- **API Integrations**  
- **JSON/CSV Files**  
- **Menus**  
- **Users**  

This would make the widgets more flexible and powerful for dynamic content creation, reducing the reliance on custom development.

### Describe the solution you'd like

I would like to see the Loop Grid and Loop Carousel widgets updated to support additional source types. Specifically:  

1. **Repeater Fields**: Similar to the functionality in the Testimonials Slider or Icon List, allow users to add and remove custom items like text, images, icons, buttons, or other widgets directly within a repeater interface.  

2. **Gallery**: Provide an option to loop through images selected in a gallery field.  

3. **API Integrations**: Allow dynamic data to be fetched from APIs and looped over for content creation.  

4. **JSON/CSV Files**: Support external file uploads or URLs to loop over rows of structured data.  

5. **Menus**: Enable looping over menu items created in WordPress menus.  

6. **Users**: Add the ability to loop over WordPress users and display dynamic user profiles.  

These features would make Loop Grid and Loop Carousel much more versatile, enabling dynamic layouts for a variety of use cases without requiring complex workarounds or custom post types.
![image](https://github.com/user-attachments/assets/aefb7e2e-6ac2-400e-9869-6e5e4584353d)

### Describe Alternatives You've Considered

Currently, the only way to loop data using repeater fields is through widgets like the Testimonials Slider or Icon List. While these widgets are effective for their specific use cases, they lack the flexibility needed for more complex or diverse dynamic content.

Another alternative is creating custom post types and fields for each type of dynamic content. However, this approach is time-intensive, especially for simpler projects, and may require additional technical expertise.

### Additional Context

Expanding the source options for Loop Grid and Loop Carousel would significantly enhance Elementor's versatility, making it easier to create dynamic layouts for a variety of use cases. This feature would save time, reduce complexity, and provide immense value to the Elementor community.  

Thank you for considering this feature request.

**Comment by pixelloop** [comment]
This would be a great feature to have.

**Comment by ghost** [comment]
+1

**Comment by FlourishDesignStudio** [comment]
This would be an excellent feature for sections like Teams, etc! Please add this feature! :)

**Comment by hassanrazadev** [comment]
+1

**Comment by razzner** [comment]
+1

**Reply by ayoubkhan558** [reply]
please upvote first comment, instead of commenting.

**Comment by sinanisler** [comment]
**editor  is laggy as hell at the moment I think they shouldnt add anything until editor itself optimized.**

latest 3 version created so many bugs I stop counting...
I am tired reporting too.....  I feel like  I am working for elementor as a tester   🤣

**Comment by RyanAlgo** [comment]
Adding an ACF Repeater function using just elementor pro without using 3rd party such as Anywhere Elementor would be great. Especially in cases like adding a specs sheet on woocommerce, teams with custom post-type, and more.  Instead of listing them one by one on icon list widget.

**Reply by felmerald** [reply]
+1 and I would highly recommend to add ACF Repeater too @ayoubkhan558

**Comment by kovou** [comment]
+1

**Comment by robstyrrell** [comment]
+1

**Comment by hiviz** [comment]
+1

**Comment by nunoliveira-empty** [comment]
+1

**Comment by nicholaszein** [comment]
Hello everyone,

I see some here referring to ACF support, while from my understanding this would be a request to use the same functionality as you can see in a few widgets, where you can add more items and define some settings for each item. 

While we understand the need from users to support ACF Pro Repeater fields, I think the request here is different (but I might be totally wrong). If you want, you can join the discussion in this thread instead if you're looking to request support for ACF Repeater fields: #5979

For this request (#22170), there are already some solutions in place. One is the nested Carousel widget. Which allows you to add any widget to each Slide just as described here.

As for the Loop Grid, the alternative would be to use a Grid Container, and simply build your inner containers with the widgets you want. The possibilities are virtually endless.

There are already solutions in place for this. While they don't work in the same way as suggested, they are simple to implement and iterate, and would lead to the same results. The idea behind the Container and Nested Elements is to give you full control over your designs and layouts without limiting to what a few settings in a widget could provide. This is obviously just a suggestion of alternatives that are already in place. Your mileage may vary. I just wanted to share them here. I hope it helps. 🙂 

Kind regards

**Reply by FlourishDesignStudio** [reply]
Hi and thanks for the response! While I understand that the Carousel is similar, it is not quite the same as what is being requested. With the original proposed repeater integration, styling happens in one place (the custom loop template). With the carousel, each segment has to be updated individually. What would be great is if you could create a loop template, and then instead of populating it with post data, it would be populated with data entered into the repeater fields (OR assigned to dynamic content). 

Sometimes you want the same styling to apply to multiple items but those items are not necessarily posts. (For example, we have these items for testimonial carousel- which are not posts but the same styling is applied to each item added in the repeater area.) 

Hope this is helpful!

**Reply by FlourishDesignStudio** [reply]
> Hello everyone,
> 
> I see some here referring to ACF support, while from my understanding this would be a request to use the same functionality as you can see in a few widgets, where you can add more items and define some settings for each item.
> 
> While we understand the need from users to support ACF Pro Repeater fields, I think the request here is different (but I might be totally wrong). If you want, you can join the discussion in this thread instead if you're looking to request support for ACF Repeater fields: #5979
> 
> For this request (#22170), there are already some solutions in place. One is the nested Carousel widget. Which allows you to add any widget to each Slide just as described here.
> 
> As for the Loop Grid, the alternative would be to use a Grid Container, and simply build your inner containers with the widgets you want. The possibilities are virtually endless.
> 
> There are already solutions in place for this. While they don't work in the same way as suggested, they are simple to implement and iterate, and would lead to the same results. The idea behind the Container and Nested Elements is to give you full control over your designs and layouts without limiting to what a few settings in a widget could provide. This is obviously just a suggestion of alternatives that are already in place. Your mileage may vary. I just wanted to share them here. I hope it helps. 🙂
> 
> Kind regards
...........................................................................................................

Hi again! I'm working on a new site today and came across another way that I would use this feature, and I thought it might help clarify... I'd like to create a team section with a custom layout. The layout would apply to each team member and I could add the fields that I would like to use with the team members. Then, I would like to use repeater fields to populate the section rather that using custom post types, which makes it easier for the client to manage. I work with 100+ clients and they ALWAYS prefer when things are editable on ONE page rather than aving to go to multiple dashboard areas to add/edit. I also prefer this!

I hope that this scenario helps to explain it a bit better!

**Reply by ayoubkhan558** [reply]
> Hello everyone,
> 
> I see some here referring to ACF support, while from my understanding this would be a request to use the same functionality as you can see in a few widgets, where you can add more items and define some settings for each item.
> 
> While we understand the need from users to support ACF Pro Repeater fields, I think the request here is different (but I might be totally wrong). If you want, you can join the discussion in this thread instead if you're looking to request support for ACF Repeater fields: #5979
> 
> For this request (#22170), there are already some solutions in place. One is the nested Carousel widget. Which allows you to add any widget to each Slide just as described here.
> 
> As for the Loop Grid, the alternative would be to use a Grid Container, and simply build your inner containers with the widgets you want. The possibilities are virtually endless.
> 
> There are already solutions in place for this. While they don't work in the same way as suggested, they are simple to implement and iterate, and would lead to the same results. The idea behind the Container and Nested Elements is to give you full control over your designs and layouts without limiting to what a few settings in a widget could provide. This is obviously just a suggestion of alternatives that are already in place. Your mileage may vary. I just wanted to share them here. I hope it helps. 🙂
> 
> Kind regards

@nicholaszein Please check the title and description. I have updated the title and description.

**Comment by xgrid-ayoubkhan558** [comment]
+1

**Comment by xgrid-ayoubkhan558** [comment]
any update?

**Comment by seamusashley** [comment]
+1 This would be super useful

**Comment by voitsekhovskymax** [comment]
+1 Update?

**Comment by Mesmer99** [comment]
Also more control for woocommerce terms will be great

**Comment by amirhp-com** [comment]
+1
